packages/hoppscotch-ui/src/components/smart/Table.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+  <div class="overflow-auto rounded-md border border-dividerDark shadow-md">
+    <table class="w-full">
+      <thead>
+        <slot name="head">
+          <tr
+            class="text-secondary border-b border-dividerDark text-sm text-left bg-primaryLight"
+          >
+            <th v-for="th in headings" scope="col" class="px-6 py-3">
+              {{ th.label ?? th.key }}
+            </th>
+          </tr>
+        </slot>
+      </thead>
+
+      <tbody class="divide-y divide-divider">
+        <slot name="body" :list="list">
```

Comment:
Is it necessary to supply the `list` as a slot prop? Isn't that something the component receives as a prop, and the parent context has access to it?

Reply:
We are using it in the parent to render individual items in the list. We are sending the list as a prop and also using it as a slot prop because in future, we might implement features such as filtering and sorting